lib/storeminator.js

```diff
@@ -12,26 +13,36 @@ module.exports = ((redis, client) => {
         getGivenCap,
     } = store(redis, client);
 
-    function storeminator(msg) {
-        const { giver, updates } = msg;
+    function handleMsg(giver, updates) {
+        getGivenCap(giver).then((res) => {
+            if (res.length >= 5) {
+                return false;
+            }
+            const a = updates.shift();
+            if (a.type === 'inc') {
+                giveBurrito(a.username);
+            } else if (a.type === 'dec') {
+                takeAwayBurrito(a.username);
+            }
 
-        getGivenCap(giver).then(() => {
-            updates.forEach((a) => {
-                if (a.type === 'inc') {
-                    giveBurrito(a.username);
-                } else if (a.type === 'dec') {
-                    takeAwayBurrito(a.username);
-                }
+            incrGiven(giver);
+            addGiver(a.username, giver);
 
-                incrGiven(giver);
-                addGiver(a.username, giver);
-                incrGivenCap(giver);
-            });
-        }).then(() => {
-            getGivenCap(giver).then(() => {
-                incrGivenCap(giver);
+            incrGivenCap(giver).then(() => {
+                if (updates.length) {
+                    handleMsg(giver, updates);
+                }
             });
+        }).catch(() => {
+            console.log('Daily cap for %s', giver);
         });
     }
+
+    function storeminator(msg) {
+        const { giver, updates } = msg;
+        if (updates.length <= 5) {
```

Comment:
Shouldn't you be allowed to give burritos until the cap is reached? This would stop all burritos if you give more than 5. But maybe this is how it's supposed to be?